src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResourceIntTest.kt

```diff
@@ -179,91 +179,192 @@ class ActivitiesResourceIntTest : IntegrationTestBase() {
+      capacity = 12,
+      startDate = LocalDate.of(2022, 10, 31),
+      endDate = LocalDate.of(2022, 11, 30),
+      minimumIncentiveLevelCode = "STD",
+      internalLocationId = -27,
+      payRates = listOf(PayRateRequest(incentiveLevel = "BAS", payBand = "5", rate = BigDecimal.valueOf(0.4)))
+    )
+
+    private fun updateActivityRequestJson(
+      prisonIdJson: String? = """"prisonId": "LEI",""",
+      capacityJson: String = """"capacity": 12,""",
+      startDateJson: String = """"startDate" : "2022-10-31",""",
+      payRatesJson: String? = """
+          "payRates" : [ {
+              "incentiveLevel" : "BAS",
+              "payBand" : "5",
+              "rate" : 0.4
+              } ]
+      """.trimIndent(),
+    ) = """{
+            ${prisonIdJson ?: ""}
+            "code" : "CA",
+            "programCode" : "SOME_PROGRAM",
+            "description" : "test description",
+            $capacityJson
+            $startDateJson
+            "endDate" : "2022-11-30",
+            "minimumIncentiveLevelCode" : "STD",
+            "internalLocationId" : -27,
+            ${payRatesJson ?: ""}
+          }"""
+
+    @Nested
+    inner class Api {
+      @Test
+      fun `access forbidden when no authority`() {
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isUnauthorized
+      }
+
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf()))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `should call the service`() {
+        doReturn(UpdateActivityResponse(prisonId = "LEI")).whenever(activityService).updateActivity(anyLong(), any())
+
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isOk
+
+        verify(activityService).updateActivity(1, updateActivityRequest())
+      }
+
+      @Test
+      fun `should return bad request`() {
+        doThrow(BadDataException("Prison not found")).whenever(activityService).updateActivity(anyLong(), any())
```

Comment:
shouldn't you just send a prison code that doesn't exist like "ZZZ"?

Reply:
Yes when I implement the back end.